data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/MultithreadedConsumer.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.consumer;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.kafka.configuration.TopicConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * * A Multithreaded helper class which helps to process the records from multiple topics in an
+ * asynchronous way.
+ */
+@SuppressWarnings("deprecation")
+public class MultithreadedConsumer implements Runnable {
+	private KafkaConsumer<String, String> plainTextConsumer = null;
+	private KafkaConsumer<String, JsonNode> jsonConsumer = null;
+	private static final Logger LOG = LoggerFactory.getLogger(MultithreadedConsumer.class);
+	private final AtomicBoolean status = new AtomicBoolean(false);
+
+	private final TopicConfig topicConfig;
+	private final Buffer<Record<Object>> buffer;
+	private String consumerId;
+	private String consumerGroupId;
+	private Properties consumerProperties;
+	private PluginMetrics pluginMetrics;
+
+	public MultithreadedConsumer(String consumerId, String consumerGroupId, Properties properties,
+								 TopicConfig topicConfig, Buffer<Record<Object>> buffer, PluginMetrics pluginMetric) {
+		this.consumerProperties = Objects.requireNonNull(properties);
+		this.consumerId = consumerId;
+		this.consumerGroupId = consumerGroupId;
+		this.topicConfig = topicConfig;
+		this.buffer = buffer;
+		this.pluginMetrics = pluginMetric;
+		this.jsonConsumer = new KafkaConsumer<>(consumerProperties);
```

Comment:
should there be only one kafka consumer based on type json/plaintext?

Reply:
but we are creating 2 KafkaConsumer objects line (27 & 28) are we going to poll only from one? may be I am missing something.

Reply:
Please address these in Bug : https://github.com/opensearch-project/data-prepper/issues/2713